lib/openhab/dsl/time/month_day.rb

```diff
@@ -8,40 +10,21 @@ module Between
       module MonthDayRange
         include OpenHAB::Log
 
-        java_import java.time.Year
-
-        # Lambdas are used to calculate the year for the month day
-        # which must happen during evaluation time to support that rules
-        # creation and evaluation for execution are done in distinct phases
-        @current_year = -> { return Year.now }
-        @next_year = -> { return Year.now.plus_years(1) }
-
-        class << self
-          attr_reader :current_year, :next_year
-        end
-
         # Creates a range that can be compared against MonthDay objects, strings
         # or anything responding to 'to_date' to see if they are within the range
         # @return Range object representing a MonthDay Range
-        # rubocop:disable Metrics/AbcSize
-        # Range method cannot be broken up cleaner
         def self.range(range)
-          logger.trace "Creating MonthDay range from #{range}"
           raise ArgumentError, 'Supplied object must be a range' unless range.is_a? Range
 
           start = MonthDay.parse(range.begin)
           ending = MonthDay.parse(range.end)
 
-          logger.trace "Month Day Range Start(#{start}) - End (#{ending})"
-
-          # Wrap to next year if ending day of month is before starting day of month
-          ending_year = ending < start ? next_year : current_year
+          start_range = DayOfYear.new(month_day: start, range: start..ending)
+          ending_range = DayOfYear.new(month_day: ending, range: start..ending)
+          logger.trace "Month Day Range Start(#{start}) - End (#{ending}) - Created from (#{range})"
 
-          start_range = MonthDayRangeElement.new(month_day: start, year: current_year)
-          ending_range = MonthDayRangeElement.new(month_day: ending, year: ending_year)
           range.exclude_end? ? (start_range...ending_range) : (start_range..ending_range)
```

Comment:
Not a suggestion, just a question for my edification: why is the above preferred to:
```ruby
Range.new(start_range, ending_range, range.exclude_end?)
```